src/common/secret.c

```diff
@@ -101,8 +101,8 @@ int get_secret_option(const char *secret, const char *key_name,
 
   option[olen] = '\0';
 
-
-  if (secret) {
+  /* when parsing kernel options (-o remount) we get '<hidden>' as the secret */
+  if (secret && (strncmp(secret, "<hidden>", 8) != 0)) {
```

Comment:
Also, I assume we know that both strings are null terminated here, so it may be better to use strcmp or maybe even memcmp. If secret is a string like "<hidden>foo", this will still return 0. Unlikely in this situation of course, but we might as well handle that situation correctly too.

Reply:
Thanks Jeff, I just pushed this change (and rebased once again to fix the make check!).